src/app/server/Server.cpp

```diff
@@ -97,7 +98,77 @@ class ServerStorageDelegate : public PersistentStorageDelegate
     }
 };
 
-ServerStorageDelegate gServerStorage;
+class ServerStorageConfig : public StorageConfiguration
+{
+public:
+    CHIP_ERROR Init(const StackParameters & parameters)
+    {
+        CHIP_ERROR err = CHIP_NO_ERROR;
+
+#if CHIP_DEVICE_LAYER_TARGET_DARWIN
+        err = PersistedStorage::KeyValueStoreMgrImpl().Init("chip.store");
```

Comment:
idea: Should we add `Init` method to the common `KeyValueStoreMgr` interface, add some "storageLocation" parameter to `StackParameters` and unify this initialization code by implementing `DefaultStorageConfiguration` when the device layer is present?

Reply:
This is a good suggestion, but this PR is only refactoring current code, I'd like it to be changed as less as possible.